pkg/agent/route/route_linux.go

```diff
@@ -214,6 +215,24 @@ func (c *Client) initIPTables() error {
 			"-i", c.hostGateway, "!", "-d", c.serviceCIDR.String(),
 			"-j", iptables.MarkTarget, "--set-xmark", "0/0xffffffff",
 		}...)
+		// When Antrea is used to enforce NetworkPolicies in EKS, an additional iptables
+		// mangle rule is required. See https://github.com/vmware-tanzu/antrea/issues/678.
+		if env.IsCloudEKS() {
+			// TODO: the following should be taking into account:
+			//   1) AWS_VPC_CNI_NODE_PORT_SUPPORT may be set to false (by default is
+			//   true), in which case we do not need to install the rule.
+			//   2) this option is not documented but the mark value can be configured
+			//   with AWS_VPC_K8S_CNI_CONNMARK.
+			// We could look for the rule added by AWS VPC CNI to the mangle table. If
+			// it does not exist, we do not need to install this rule. If it does exist
+			// we can scan for the mark value and use that in our rule.
+			writeLine(iptablesData, []string{
```

Comment:
Do you think cleaner to put these lines in a separate func?

Reply:
I will move it.

Reply:
Done. PTAL.